### ISSUE TITLE
Changed `Chord` type to a discriminated union

### ISSUE THREAD

**Issue by darrylnoakes** [no-chris/chord-symbol#592]
Could the `Chord` type be changed to a union? Preferably made up of two named subtypes (e.g., `ChordData` and `ChordParseError`).
Then, instead of checking every property I want to use, I can simply check whether the `error` property is present or not and the type will be narrowed.

**Comment by no-chris** [comment]
Hello,
I'm not familiar with Typescript so this is a bit confusing for me. Can you point out some documentation where I can better understand what you are asking? Thanks

**Comment by no-chris** [comment]
better yet: feel free to submit a PR, if it makes sense for TS users I'll be happy to merge it

**Comment by darrylnoakes** [comment]
I think this CSS Tricks article explains it all very well: [Demystifying TypeScript Discriminated Unions](https://css-tricks.com/typescript-discriminated-unions/).

But I will see if I can get around to making a PR :)

**Comment by darrylnoakes** [comment]
I have made three changes, but I don't know of any easy way to do a stacked PR on someone else's repo, so I'm going to have to make the PR's one at a time.

**Comment by no-chris** [comment]
Thanks! You can just add more commits to your forked branch and the PR will update automatically

**Comment by no-chris** [comment]
In any case I merged the first PR, thanks again!

**Comment by darrylnoakes** [comment]
> Thanks! You can just add more commits to your forked branch and the PR will update automatically

So you wouldn't mind me lumping them all together in one PR?

**Comment by no-chris** [comment]
Yes, as long as they all related to the TS file definition I'm fine with a single PR

**Comment by darrylnoakes** [comment]
I have made the PR for the chord object typing.

The other change I made was improving (maybe) and exporting the custom filter type, which makes it easier to type custom filter functions that are not defined directly in the config. I also typed the renderer filter in the Ultimate Guitar package.
As this could be a matter of opinion, I decided to leave it for a separate PR that can be modified and/or rejected apart from the one for the chord object typing.
